pages/auth/authorize.vue

```diff
@@ -0,0 +1,308 @@
+        </div>
+      </div>
+      <div class="title">
+        <h1>Authorize {{ app.name }}</h1>
+      </div>
+      <div class="auth-info">
+        <div class="scope-heading">
+          <strong>{{ app.name }}</strong> by
+          <nuxt-link class="text-link" :to="'/user/' + createdBy.id">{{
+            createdBy.username
+          }}</nuxt-link>
+          will be able to:
+        </div>
+        <div class="scope-items">
+          <div v-for="scopeItem in scopeDefinitions" :key="scopeItem">
+            <div class="scope-item">
+              <div class="scope-icon">
+                <CheckIcon />
+              </div>
+              {{ scopeItem }}
+            </div>
+          </div>
+        </div>
+      </div>
+      <div class="button-row">
+        <Button class="wide-button" large :action="onReject" :disabled="pending">
+          <XIcon />
+          Decline
+        </Button>
+        <Button class="wide-button" color="primary" large :action="onAuthorize" :disabled="pending">
+          <CheckIcon />
+          Authorize
+        </Button>
+      </div>
+      <div class="redirection-notice">
+        <p class="redirect-instructions">
+          You will be redirected to
+          <span class="redirect-url">{{ redirectUri }}</span>
+        </p>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup>
+import { Button, XIcon, CheckIcon, Avatar } from 'omorphia'
+import { useBaseFetch } from '@/composables/fetch.js'
+import { useAuth } from '@/composables/auth.js'
+import { getScopeDefinitions } from '@/utils/auth/scopes.ts'
+
+const data = useNuxtApp()
+
+const router = useRoute()
+const auth = await useAuth()
+
+const clientId = router.query?.client_id || false
+const redirectUri = router.query?.redirect_uri || false
+const scope = router.query?.scope || false
+const state = router.query?.state || false
+
+const getFlowIdAuthorization = async () => {
+  const query = {
+    client_id: clientId,
+    redirect_uri: redirectUri,
+    scope,
+  }
+  if (state) {
+    query.state = state
+  }
+
+  const authorization = await useBaseFetch('oauth/authorize', {
+    method: 'GET',
+    apiVersion: 3,
+    query,
+  }) // This will contain the flow_id and oauth_client_id for accepting the oauth on behalf of the user
+
+  if (typeof authorization === 'string') {
+    await navigateTo(authorization, {
+      external: true,
+    })
+  }
+
+  return authorization
+}
+
+const {
+  data: authorizationData,
+  pending,
+  error,
+} = await useAsyncData('authorization', getFlowIdAuthorization)
+
+const app = await useBaseFetch('oauth/app/' + clientId, {
+  method: 'GET',
+  apiVersion: 3,
+})
+
+const scopeDefinitions = getScopeDefinitions(BigInt(authorizationData.value?.requested_scopes || 0))
+
+const createdBy = await useBaseFetch('user/' + app.created_by, {
+  method: 'GET',
+})
+
+const onAuthorize = async () => {
+  try {
+    const res = await useBaseFetch('auth/oauth/accept', {
```

Comment:
This needs to be changed to `oauth/accept` - API route changed.

```suggestion
    const res = await useBaseFetch('oauth/accept', {
```